src/main/java/bio/terra/workspace/service/datareference/DataReferenceService.java

```diff
@@ -128,6 +132,24 @@ public DataReference createDataReference(
     return dataReferenceDao.enumerateDataReferences(workspaceId, offset, limit);
   }
 
+  @Traced
+  public void updateDataReference(
+      UUID workspaceId,
+      UUID referenceId,
+      UpdateDataReferenceRequestBody updateRequest,
+      AuthenticatedUserRequest userReq) {
+    workspaceService.validateWorkspaceAndAction(
+        userReq, workspaceId, SamConstants.SAM_WORKSPACE_WRITE_ACTION);
+
+    if (!dataReferenceDao.updateDataReference(
+        workspaceId,
+        referenceId,
+        updateRequest.getName(),
+        updateRequest.getReferenceDescription())) {
+      throw new DataReferenceNotFoundException("Data Reference not found.");
```

Comment:
I think you can also reach here if both name and reference are null; `DataReferenceNotFoundException` is misleading in that case.

Reply:
Adding a guard at the controller level.